src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -1725,16 +1725,13 @@ private bool ShouldReduceSizeOfCompressionPager(long initialSize, bool forceRedu
             if ((DateTime.UtcNow - _lastCompressionBufferReduceCheck).TotalMinutes < 5)
                 return false;
 
-            if (_maxNumberOfPagesRequiredForCompressionBuffer > _compressionPager.NumberOfAllocatedPages / 2)
-            {
-                // we recently used at least half of the compression buffer,
-                // no point in reducing size yet, we'll try to do it again next time
-                return false;
-            }
+            
+            // prevent resize if we recently used at least half of the compression buffer
+            var preventResize = _maxNumberOfPagesRequiredForCompressionBuffer > _compressionPager.NumberOfAllocatedPages / 2;
```

Comment:
That would still cause "bouncing". If you transactions are varying in size, that can cause you to grow & shrink things rapidly.
The problem is that we only have `initialSize` here, and we need the `maxSize` as well

Reply:
The bounce would (with the latest change) happen only all five minutes, but I get your point. Should I close the PR, or attempt a "real" fix sometimes this week?

Reply:
I see you are way ahead of me; I see no more issues with your latest commit